README.md

```diff
@@ -1,335 +1,446 @@
 
-Any Puppet code that uses anything from the apt module requires that the core apt class be declared.
+~~~
+include apt
+~~~
+
+**Note:** The main `apt` class is required by all other classes, types, and defines in this module. You must declare it whenever you use the module.
 
 ## Usage
 
-Using the apt module consists predominantly of declaring classes and defined types that provide the desired functionality and features. This module provides common resources and options that are shared by the various defined types in the apt module, so you **must always** include this class in your manifests.
+### Add GPG keys
+
+**Warning:** Using short key IDs presents a serious security issue, potentially leaving you open to collision attacks. We recommend you always use full fingerprints to identify your GPG keys. This module allows short keys, but issues a security warning if you use them.
+
+Declare the `apt::key` class:
+
+~~~
+apt::key { 'puppetlabs':
+  id      => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+  server  => 'pgp.mit.edu',
+  options => 'http-proxy="http://proxyuser:proxypass@example.org:3128"',
+}
+~~~
+
+You can make Apt load your key before others by adjusting the `priority` parameter (the default priority is 50).
+
+~~~
+apt::key { 'puppetlabs':
+  id       => '47B320EB4C7C375AA9DAE1A01054B7A24BD6EC30',
+  server   => 'pgp.mit.edu',
+  options  => 'http-proxy="http://proxyuser:proxypass@example.org:3128"',
+  priority => '50',
+}
+~~~
+
+### Prioritize backports
+
+~~~
+class { 'apt::backports':
+  pin => 500,
+}
+~~~
+
+By default, the `apt::backports` class drops a pin file for backports, pinning it to a priority of 200. This is lower than the normal default of 500, so packages with `ensure => latest` don't get upgraded from backports without your explicit permission.
+
+If you raise the priority through the `pin` parameter to 500, normal policy goes into effect and Apt installs or upgrades to the newest version. This means that if a package is available from backports, it and its dependencies are pulled in from backports unless you explicitly set the `ensure` attribute of the `package` resource to `installed`/`present` or a specific version.
+
+### Update the list of packages
+
+By default, Puppet runs `apt-get update` on the first Puppet run after you include the `apt` class, and anytime `notify  => Exec['apt_update']` occurs; i.e., whenever config files get updated or other relevant changes occur. If you set `update['frequency']` to 'always', the update runs on every Puppet run. You can also set `update['frequency']` to 'daily' or 'weekly':
+
+~~~
+class { 'apt':
+  update => {
+    frequency => 'daily',
+  },
+}
+~~~
+
+### Pin a specific release
+
+~~~
+apt::pin { 'karmic': priority => 700 }
+apt::pin { 'karmic-updates': priority => 700 }
+apt::pin { 'karmic-security': priority => 700 }
+~~~
+
+You can also specify more complex pins using distribution properties:
+
+~~~
```

Comment:
Can we please use triple back ticks \` and add \`\`\`puppet\` to it so we get syntax highlighting?